lib/gist.rb

```diff
@@ -136,9 +136,9 @@ def multi_gist(files, options={})
 
     url = "#{base_path}/gists"
     url << "/" << CGI.escape(existing_gist) if existing_gist.to_s != ''
-    url << "?access_token=" << CGI.escape(access_token) if access_token.to_s != ''
 
     request = Net::HTTP::Post.new(url)
+    request['Authorization'] = "token #{access_token}" if access_token.to_s != ''
```

Comment:
@keynslug thanks for this change!

In general it's unsafe to echo user-provided input into HTTP headers (as if the input contains `\r\n` then they can control more than just one header). The risk is very low in this context (as the access token is already assumed to be trusted), but as good practice I think we should do something like:

`"token #{header_safe access_token}"` where 

```
def header_safe(x) 
   raise "invalid access_token" if x =~ /[\r\n]/
   x
end
```

Reply:
As Conrad notes, this is probably needed, but the input reader presently stops this anyway, so it's only an extra security step (a very good one).

Reply:
Oh. I had an impression that it's already taken care of by these lines of code from `Net::HTTPHeader`:

https://github.com/ruby/ruby/blob/v2_7_0/lib/net/http/header.rb#L84-L86

Reply:
Perfect, I was not aware of that!